### DISCUSSION TITLE
any plans to make an api for body spawning and deleting?

### DISCUSSION THREAD

**Discussion by MyelinsheathXD** [google-deepmind/mujoco#154] [Ideas]
Hey! It's great to see that such an incredible simulation in real time.

But, with a gaming perspective,  a dynamic changes at runtime is crucial.

Is there any tricks to modify "mjdata" data structure manually and create and delete bodies at runtime?

**Comment by erez-tom** [comment]
The model needs to be recompiled after bodies are spawned or deleted.  Our Unity plug-in will trigger recompilation automatically when MuJoCo components are added or removed, and the state of persisting bodies will be preserved across the recompilation, see implementation [here](https://github.com/deepmind/mujoco/blob/main/unity/Runtime/Components/MjScene.cs#L198).  If you wish to recreate this behaviour outside Unity, you can use a similar pattern for carrying the state across.
